velox/functions/lib/string/StringImpl.h

```diff
@@ -162,6 +162,95 @@ FOLLY_ALWAYS_INLINE int32_t charToCodePoint(const T& inputString) {
   return codePoint;
 }
 
+template <typename TInString>
+FOLLY_ALWAYS_INLINE int64_t
+hammingDistance(const TInString& left, const TInString& right) {
+  int64_t distance = 0;
+  int64_t leftPosition = 0;
+  int64_t rightPosition = 0;
+  std::string_view leftSV = std::string_view(left.data(), left.size());
+  std::string_view rightSV = std::string_view(right.data(), right.size());
+
+  while (leftPosition < leftSV.size() && rightPosition < rightSV.size()) {
+    int size = 0;
+    auto codePointLeft = utf8proc_codepoint(&leftSV.at(leftPosition), size);
+    leftPosition += size;
+    auto codePointRight = utf8proc_codepoint(&rightSV.at(rightPosition), size);
+    rightPosition += size;
+
+    if (codePointLeft != codePointRight) {
+      distance++;
+    }
+  }
+
+  VELOX_USER_CHECK(
+      leftPosition == leftSV.size() && rightPosition == rightSV.size(),
+      "The input strings to hamming_distance function must have the same length");
+
+  return distance;
+}
+
+FOLLY_ALWAYS_INLINE void castToCodePoints(
+    std::string_view str,
+    std::vector<int32_t>& codePoints) {
+  for (int index = 0; index < str.size();) {
+    int size = 0;
+    auto codePoint = utf8proc_codepoint(&str.at(index), size);
+    VELOX_USER_CHECK(
+        codePoint > 0, "Invalid UTF-8 encoding in characters: {}", str.data());
+    index += utf8proc_codepoint_length(codePoint);
```

Comment:
Is this different from the value stored in size ?

Reply:
They would be different only if the codepoint is invalid. In this case we can reuse size